openoa/utils/entr/__init__.py

```diff
@@ -415,6 +418,210 @@ def load_reanalysis(conn:EntrConnection, plant_metadata:dict, reanalysis_product
     return reanalysis_df_dict, reanalysis_meta_dict
 
 
+### Multi-Plant AEP Map
+
+def aep_spark_map_udf(aep_spark_map_udf_broadcast_metadata, plant_name, df):
+    """
+    Pyspark UDF parial, applied via applyInPandas
+    """
+    from openoa.analysis import MonteCarloAEP
+    from openoa.plant import PlantData
+
+    fct_pandas = df
+    plant_name = plant_name[0] # Spark assigns group key as a tuple, which would generalize to multiple group keys
+
+    #global aep_spark_map_udf_broadcast_metadata
+    plant_meta = aep_spark_map_udf_broadcast_metadata.value[plant_name]
+
+    openoa_era5 = fct_pandas[(fct_pandas["table"]=="reanalysis") &\
+                               (fct_pandas["reanalysis_dataset_name"]=="ERA5")]\
+                            .pivot(index="date_time", columns="entr_tag_name", values="tag_value")\
+                            .reset_index()
+
+    # OpenOA MERRA2 table
+    openoa_merra2 = fct_pandas[(fct_pandas["table"]=="reanalysis") &\
+                               (fct_pandas["reanalysis_dataset_name"]=="MERRA2")]\
+                            .pivot(index="date_time", columns="entr_tag_name", values="tag_value")\
+                            .reset_index()
+
+    # OpenOA Curtail table
+    # TODO: We are resampling the availability and curtailment values to monthly. This prevents an index error in OpenOA when these variables have different frequencies.
+    openoa_curtail = fct_pandas[(fct_pandas["table"]=="plant") &\
+                               (fct_pandas["entr_tag_name"].isin(["IAVL.DnWh", "IAVL.ExtPwrDnWh"]))]\
+                            .pivot(index="date_time", columns="entr_tag_name", values="tag_value")\
+                            .resample("1M").sum()\
+                            .reset_index()
```

Comment:
Here or in a later version it would be good to resample to the lowest resolution out of the availability and curtailment data instead of always monthly, so we're not forced to use monthly resolution in the AEP analysis if the data is higher resolution.